### PR TITLE
Adding ColorColumn (companion for ColorPicker form field)

### PR DESCRIPTION
**Pull request by cheesegrits**
As agreed with @saade, recreated ColorColumn from the deprecated colorpicker plugin to accompany the core ColorPicker field.

### CODE REVIEW

**Comment by cheesegrits** [comment]
I'm not wild about the way the "Copied" message shows after clicking in copyable() mode (shifts the swatch up and shows a simple div for configurable amount of time), but I'm not a UX guy, I just copied the way the original (deprecated) plugin did it.  If you have suggestions for a better way to do it, lmk and I'll implement it.

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                            }, {{ $getCopyMessageShowTimeMs() }});
+                        });
+                },
+            }"
+            @click.prevent.stop="copy"
+        >
+            @endif
+            <span
+                class="relative flex"
+                style="background-color: {{ $getState() }}; width: 1.5rem; height: 1.5rem; margin-left: 1rem; border-radius: 6px;"
```
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:26]: Could you move these inline styles (except the background color of course) to Tailwind classes please?

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by zepfietje** [comment]
As for the copy behavior, I'm thinking this can be achieved quite easily by using `extraAttributes()` and sending a notification from the cell action. I'm just wondering if we should have the copy behavior in core (as there's plugins for other copy columns), though it makes sense for a color column.

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                title="{{ $getState() }}"
+            >
+            </span>
+            @if ($isCopyable())
```
Comment (danharrin) [inline packages/tables/resources/views/columns/color-column.blade.php:30]: We can use the tooltip package for this message - https://github.com/ryangjchandler/alpine-tooltip#tooltip

Comment (cheesegrits) [reply packages/tables/resources/views/columns/color-column.blade.php:30]: OK, I'll fix that

File: packages/tables/resources/views/columns/color-column.blade.php (hunk 1 above)
Comment (cheesegrits) [reply packages/tables/resources/views/columns/color-column.blade.php]: This one is resolved.

File: packages/tables/resources/views/columns/color-column.blade.php (hunk 2 above)
Comment (cheesegrits) [reply packages/tables/resources/views/columns/color-column.blade.php]: Done

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            x-data="{
+                showCopied: false,
+                copiedMessageTimeout: -1,
+                copy() {
+                    window
+                        .navigator
+                        .clipboard
+                        .writeText('{{ $getState() }}');
+                },
+            }"
```
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:13]: This is no longer needed?

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            @click="$tooltip('{{ $getCopyMessage() }}', { timeout: {{ $getCopyMessageShowTimeMs() }} })"
+            @click.prevent.stop="copy"
+        >
+    @endif
+            <span
+                {{ $attributes->merge($getExtraAttributes())->class([
+	                'filament-tables-color-column relative flex w-6 h-6 ml-4 rounded-md'
+	            ]) }}
```
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:21]: Please change the tabs to spaces in this indentation.

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                style="background-color: {{ $getState() }};"
+                title="{{ $getState() }}"
+            >
+            </span>
```
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:25]: Can't the span and button be merged?

File: packages/tables/resources/views/columns/color-column.blade.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    @if ($isCopyable())
+        </button>
+    @endif
+@else
+    <span>&nbsp;</span>
```
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:30]: What's this for?

File: packages/tables/resources/views/columns/color-column.blade.php (hunk 4 above)
Comment (zepfietje) [inline packages/tables/resources/views/columns/color-column.blade.php:21]:
```suggestion
                {{
                    $attributes
                        ->merge($getExtraAttributes())
                        ->class(['filament-tables-color-column relative flex w-6 h-6 ml-4 rounded-md'])
                }}
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by zepfietje** [comment]
I did a quick rework. @cheesegrits could you check that everything still works as expected?

@danharrin, the copy message and duration should probably also support `Closure` and `null`?

**Comment by cheesegrits** [comment]
@zepfietje I'll add closures to those, and check they handle null gracefully.

**Comment by cheesegrits** [comment]
@zepfietje OK, closures and null handling added.

BTW, the reason I'd left it using a button when copyable was so you get the visual mouse pointer change on hover, indicating that clicking "does something".  I just tried adding some Tailwind hover pseudo events, but they don't seem to do anything.  Not a big deal, just let me know if you have opinions either way on this.

Oh, and I changed trans() to __().  I have no idea what the difference is, but seems like you use __() everywhere else.

File: packages/tables/src/Columns/ColorColumn.php [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    public function copyMessageDuration(int | Closure | null $duration): self
+    {
+        $this->copyMessageDuration = $duration;
+
+        return $this;
+    }
+
+    public function getCopyMessageDuration(): ?int
+    {
+        return $this->evaluate($this->copyMessageDuration ?? 0);
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:45]:
```suggestion
        return $this->evaluate($this->copyMessageDuration) ?? 2000;
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public function copyMessageDuration(int | Closure | null $duration): self
+    {
+        $this->copyMessageDuration = $duration;
+
+        return $this;
+    }
+
+    public function getCopyMessageDuration(): ?int
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:43]:
```suggestion
    public function getCopyMessageDuration(): int
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+        return $this;
+    }
+
+    public function getCopyMessage(): ?string
+    {
+        return $this->evaluate($this->copyMessage ?? __('tables::table.columns.color.copied'));
+    }
+
+    public function copyMessageDuration(int | Closure | null $duration): self
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:36]:
```suggestion
    public function copyMessageDuration(int | Closure | null $duration): static
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    public function copyMessage(string | Closure | null $message): self
+    {
+        $this->copyMessage = $message;
+
+        return $this;
+    }
+
+    public function getCopyMessage(): ?string
+    {
+        return $this->evaluate($this->copyMessage ?? __('tables::table.columns.color.copied'));
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:33]:
```suggestion
        return $this->evaluate($this->copyMessage) ?? __('tables::table.columns.color.messages.copied');
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public function copyMessage(string | Closure | null $message): self
+    {
+        $this->copyMessage = $message;
+
+        return $this;
+    }
+
+    public function getCopyMessage(): ?string
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:31]:
```suggestion
    public function getCopyMessage(): string
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    protected int | Closure | null $copyMessageDuration = 2000;
+
+    public function copyable(bool | Closure $copyable = true): self
+    {
+        $this->isCopyable = $copyable;
+
+        return $this;
+    }
+
+    public function copyMessage(string | Closure | null $message): self
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:24]:
```suggestion
    public function copyMessage(string | Closure | null $message): static
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+{
+    protected string $view = 'tables::columns.color-column';
+
+    protected bool | Closure $isCopyable = false;
+
+    protected string | Closure | null $copyMessage = null;
+
+    protected int | Closure | null $copyMessageDuration = 2000;
+
+    public function copyable(bool | Closure $copyable = true): self
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:17]:
```suggestion
    public function copyable(bool | Closure $condition = true): static
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        $this->isCopyable = $copyable;
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:19]:
```suggestion
        $this->isCopyable = $condition;
```

File: packages/tables/src/Columns/ColorColumn.php [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+class ColorColumn extends Column
+{
+    protected string $view = 'tables::columns.color-column';
+
+    protected bool | Closure $isCopyable = false;
+
+    protected string | Closure | null $copyMessage = null;
+
+    protected int | Closure | null $copyMessageDuration = 2000;
```
Comment (danharrin) [inline packages/tables/src/Columns/ColorColumn.php:15]:
```suggestion
    protected int | Closure | null $copyMessageDuration = null;
```

File: packages/tables/resources/lang/en/table.php [hunk 16]
```diff
@@ -8,6 +8,9 @@
             'more' => 'and :count more',
         ],
 
+        'color' => [
+            'copied' => 'Copied',
```
Comment (danharrin) [inline packages/tables/resources/lang/en/table.php:12]:
```suggestion

            'messages' => [
                'copied' => 'Copied',
            ],
            
```

File: packages/tables/docs/03-columns.md [hunk 17]
```diff
@@ -845,6 +845,27 @@ use Filament\Tables\Columns\ToggleColumn;
 ToggleColumn::make('is_admin')
 ```
 
+## Color column
+
+The color column allows you to show the color preview from a CSS color definition, typically entered using the ColorPicker form field, in one of the supported formats (hex, hsl, rgb, rgba).
```
Comment (danharrin) [inline packages/tables/docs/03-columns.md:850]:
```suggestion
The color column allows you to show the color preview from a CSS color definition, typically entered using the color picker field, in one of the supported formats (HEX, HSL, RGB, RGBA).
```

File: packages/tables/docs/03-columns.md [hunk 18]
```diff
@@ -845,6 +845,27 @@ use Filament\Tables\Columns\ToggleColumn;
...
+
+```php
+use Filament\Tables\Columns\ColorColumn
+
+ColorColumn::make('color')
+```
+
+You may make the color copyable, such that clicking on the preview it copies the CSS value to the clipboard, and optionally specify the copy confirmation message (default is 'Copied') and popup message timeout in MS (default 2000).  Copying only works on `https://` environments.
```
Comment (danharrin) [inline packages/tables/docs/03-columns.md:858]:
```suggestion
You may make the color copyable, such that clicking on the preview it copies the CSS value to the clipboard, and optionally specify a custom confirmation message and duration in milliseconds.  This feature only works when SSL is enabled for the app.
```

**Review by danharrin — CHANGES_REQUESTED** [review]

**Pull request merged**